src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/ResourceIdentifier.cs

```diff
@@ -138,10 +138,16 @@ public override string ToString()
 
             AppendIfNotNull(ref resourceId, "/subscriptions/{0}", Subscription);
             AppendIfNotNull(ref resourceId, "/resourceGroups/{0}", ResourceGroupName);
-            AppendIfNotNull(ref resourceId, "/providers/{0}", provider);
-            AppendIfNotNull(ref resourceId, "/{0}", parentAndType);
-            AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
 
+            // Only append parentAndType and ResoruceName when provider is not empty.
+            // Otherwise the url will be invalid.
+            if (!string.IsNullOrEmpty(provider))
```

Comment:
also, take a  look at the constructor for this class.  is this class called from new role assignment cmdlet?

---
In reply to: [118149831](https://github.com/Azure/azure-powershell/pull/4019#discussion_r118149831) [](ancestors = 118149831)